client-app/src/Components/DonatedItemsList.tsx

```diff
@@ -0,0 +1,527 @@
+import React, { useState, useEffect } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { FaSearch, FaPlus } from 'react-icons/fa';
+import Modal from 'react-modal';
+import '../css/AdminHeader.css';
+import '../css/DonatedItemsList.css';
+import html2canvas from 'html2canvas';
+import Barcode from 'react-barcode';
+
+// Define interfaces for our data structures
+interface Donor {
+    id: number;
+    firstName: string;
+    email: string;
+    name?: string;
+}
+
+interface Program {
+    id: number;
+    name: string;
+}
+
+interface Status {
+    status: string;
+    date: string;
+}
+
+interface DonatedItem {
+    id: number;
+    itemType: string;
+    currentStatus: string;
+    dateDonated: string;
+    lastUpdated?: string;
+    donor: Donor | null;
+    program: Program | null;
+    programId?: number;
+    statuses?: Status[];
+}
+
+interface SelectedItemDetails extends DonatedItem {
+    statuses: Status[];
+}
+
+const DonatedItemsList: React.FC = () => {
+    const [searchInput, setSearchInput] = useState<string>('');
+    const [filteredItems, setFilteredItems] = useState<DonatedItem[]>([]);
+    const [selectedItems, setSelectedItems] = useState<number[]>([]);
+    const [selectedItemDetails, setSelectedItemDetails] =
+        useState<SelectedItemDetails | null>(null);
+
+    const [donatedItems, setDonatedItems] = useState<DonatedItem[]>([]);
+    const [loading, setLoading] = useState<boolean>(true);
+    const [error, setError] = useState<string | null>(null);
+    const [programOptions, setProgramOptions] = useState<Program[]>([]);
+    const [selectedProgram, setSelectedProgram] = useState<string>('');
+    const [assignProgramClicked, setAssignProgramClicked] =
+        useState<boolean>(false);
+    const [modalIsOpen, setModalIsOpen] = useState<boolean>(false);
+    const [itemTypes, setItemTypes] = useState<Set<string>>(new Set());
+
+    const navigate = useNavigate();
+
+    const fetchDonatedItems = async (): Promise<void> => {
+        try {
+            setLoading(true);
+            const response = await fetch(
+                `${process.env.REACT_APP_BACKEND_API_BASE_URL}/donatedItem`,
```

Comment:
Consider wrapping this and similar functions with useCallback to prevent unnecessary re-renders, especially if these handlers are passed as props to child components.

Reply:
Remove the extra `'/' ` before donatedItem in the above URL